web/src/TacComponents.tsx

```diff
@@ -487,3 +498,48 @@ export function LocatorNotification() {
     </Alert>
   );
 }
+
+export function PowerFailNotification() {
+  const state = useMqttSubscription<OutputState>("/v1/dut/powered");
+
+  let reason = null;
+
+  switch (state) {
+    case OutputState.InvertedPolarity:
+      reason = "an inverted polarity event";
+      break;
+    case OutputState.OverCurrent:
+      reason = "an overcurrent event";
+      break;
+    case OutputState.OverVoltage:
+      reason = "an overvoltage event";
+      break;
+    case OutputState.RealtimeViolation:
+      reason = "a realtime violation";
+      break;
+  }
```

Comment:
Nit: maybe we should add a default handler in case none of the cases match. Otherwise looks good to me.

Reply:
I initially agreed but I've had a second look and think we shouldn't. In the default case we would want `reason` to be `null` to make sure the `Alert` is set to be not visible:

https://github.com/linux-automation/tacd/blob/eb97382b14cbbab2e6cba0dd9ca4ac118187368d/web/src/TacComponents.tsx#L555-L562

This is however already the case when none of the `case`s match, as `reason` is initialized that way.